makeflow/src/dag_node.c

```diff
@@ -223,6 +415,607 @@ void dag_node_add_target_file(struct dag_node *n, const char *filename, char *re
 	target->created_by = n;
 }
 
+/* Function that calculates the three different footprint values and
```

Comment:
What are those three different footprints?

I would add a detailed description of the algorithm somewhere in this file, referring to functions as appropriate.